react/week3/my-app/src/render.js

```diff
@@ -0,0 +1,189 @@
+import React from "react";
+
+function TodoItem(props) {
+  const isComplete = props.todos.completed;
+  return (
+    <li
+      id="lists"
+      style={{ textDecoration: isComplete ? "line-through" : "none" }}
+    >
+      {props.todos.description} || {props.todos.deadline}
+      <input type="checkbox" onChange={props.handelCheckbox}></input>
+      <button onClick={props.deleteTodo}> Delete</button>
+      <button onClick={props.editTodo}>Edit </button>
+      <button onClick={props.updateTodo}> update </button>
+    </li>
+  );
+}
+
+class TodoList extends React.Component {
+  state = {
+    listOfTodos: [],
+    todoName: "",
+    deadlineDate: "",
+    updateInput: "",
+  };
+
+  addTodo = this.addTodo.bind(this);
+  removeTodo = this.removeTodo.bind(this);
+  handelCheckbox = this.handelCheckbox.bind(this);
+  update = this.update.bind(this);
+
+  feachTodoArray() {
+    return fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    ).then((data) => data.json());
+  }
+
+  addTodo() {
```

Comment:
Why do you have an `addTodo` method which fetches the data again from the API? The flow of this app should be something like:

- fetch data when the component did mount
- add/remove/update todos